### PR TITLE
fix(processing): place each aggregate child once in the quick-metadata spatial tree

### PR DESCRIPTION
**Pull request by louistrue**
## Defect class

The quick-metadata bootstrap spatial tree emits a node once per path that reaches it, so a malformed `IfcRelAggregates` blows the tree up exponentially (processing orchestrator review, finding 6). Split out of #4590.

## Finding

The aggregate wiring in `process_geometry_streaming_filtered_with_options` pushes every child a relationship lists, and `parse_step_ref_list` returns duplicates verbatim. `build_quick_spatial_tree_node` only guarded the current root-to-node path. That stops a back-edge but never a sibling repeat or a second parent, so a relationship naming a child k times per level produced k^depth nodes.

Measured on the test fixture with the fix reverted: the site appeared 3 times, the building 9 times and the storey 30 times.

## Fix

`rust/processing/src/processor/quick_metadata.rs`: the builder keeps one set of every node emitted anywhere in the tree and skips a child that is already in it. Each node is emitted once, where the depth-first walk from the root first reaches it, so the output is bounded by the node count. The same set still stops back-edges.

`has_children` now describes the children actually emitted. A node whose only listed children were placed elsewhere no longer advertises an empty subtree.

### Why not the suggested first-placement-wins guard in the wiring loop

That was the first version on this branch. `/code-review` found that it drops whole branches. A malformed back-edge listed before the real parent (building aggregates site, then project aggregates site: the #4246 shape) claims the site for the building. The project's relationship is then skipped, so the site, the building, the storey and their elements all fall out of the tree. The builder walks from the root, so it keeps the placement anchored at the project. The mutation table below shows that shape failing the test.

## Tests

- `quick_metadata_aggregate_dedupe::duplicated_aggregate_children_are_placed_once` (new). It appends a site/building/storey tree to the determinism fixture with:
  - each child listed three times per level
  - a back-edge before the project's relationship
  - a second parent for the storey

  It asserts that each node appears once, that the chain is project > site > building > storey, and that the storey keeps its four contained elements.
- `quick_metadata::tests::cyclic_aggregate_graph_does_not_stack_overflow` now also asserts that the node with only a back-edge child reports `has_children == false`.

Mutations, each run and each failing:

| Mutation | Result |
|---|---|
| builder reverted to main | `node #900 must be placed exactly once`, left 3 |
| placed set made path-scoped again (`placed.remove` after the child loop) | left 3 |
| old builder plus the first-placement-wins wiring guard | back-edge case, left 0 |
| `has_children` from the raw child list | the unit test fails |

## Deferred, on record

- **Recursion depth.** `build_quick_spatial_tree_node_inner` is still unbounded in depth. An acyclic chain of about 100k nested spatial nodes overflows the stack. That is a depth bound, not a repeat bound, so it is a separate defect class for its own PR. `QuickMetadataSpatialNode`'s derived `Clone`, `Serialize` and `Drop` recurse too.
- **Upstream placement guards.** The containment loop's `already_placed` guard and the root fallback's `parent.is_none()` both read `parent`, and any aggregate edge sets it, including a back-edge or an edge from an unreachable node. Not changed by this PR.
- **Test helper duplication.** The bootstrap-capture helper exists in five other test files. Moving it to `tests/common` is a separate refactor.

## Review

`/simplify` ran twice. The second run was on the builder version, and it applied the `insert`-based check, `with_capacity`, trimmed comments, and moved a misplaced test comment back onto its test. `/code-review` then ran against `284a32045` and found no defects.

One minor point from it is dismissed: it said the `#909` back-edge "would also pass on main". That is true. The back-edge case exists to catch the wiring-guard design, and the mutation table shows it does (left 0).

Since then the branch has only been rebased onto main. `git range-diff` shows the patch identical (`284a32045 = bc6a96279`).

## Gates (on `bc6a96279`, rebased on `c42d87d8b`)

- `cargo clippy --workspace --all-targets -- -D warnings`: 0
- `cargo test -p ifc-lite-processing --lib`: 398 passed
- `--test quick_metadata_aggregate_dedupe`, `issue_1075_spatial_zone`, `issue_3275_quick_spatial_types`, `issue_1910_bootstrap_regression`, `step_keyword_case_class`: all pass
- `module_size_ratchet`, `check-module-size`, `check-refwalk-guards`, `check-rust-api-surface`, `check-rust-major-offset`: all 0
- Geometry watertightness census (`triangulation_invariance --features triangulation-alt`, fixtures verified): exit 0, 65 passed, 2 ignored
- The full workspace test run was stopped on the coordinator's instruction (machine-wide binary-scan bottleneck). CI's Rust tests lane covers it.

No hot path changed: the builder runs once per bootstrap, and only when `emit_quick_metadata_bootstrap` is set.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: rust/processing/src/processor/quick_metadata.rs [hunk 1]
```diff
@@ -167,38 +167,37 @@ pub(super) fn build_quick_spatial_tree_node(
...
         .get(&express_id)
         .ok_or_else(|| format!("Quick spatial node #{express_id} not found"))?;
-    ancestors.insert(express_id);
     let mut children = Vec::with_capacity(node.children.len());
     for child_id in &node.children {
-        if ancestors.contains(child_id) {
-            // Cyclic aggregate edge: skip this back-edge child, keep the rest.
+        if !placed.insert(*child_id) {
             continue;
         }
```
[the remark this answers is not in the document]
Comment (louistrue) [reply rust/processing/src/processor/quick_metadata.rs:193]: Deferred to #4662. The finding is right that pruned edges leave no trace, but `QuickMetadataBootstrap` (rust/processing/src/types/response.rs) has no diagnostics or warnings field, only `schema_version`, `entity_count` and `spatial_tree`. Counting pruned edges there means adding a field to a public, serde-serialised type of the published crate, which is outside this PR's defect class (the k^depth blow-up). #4662 records the proposal: a pruned-edge count with a serde default, plus a test on this PR's fixture.
